superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx

```diff
@@ -68,47 +222,56 @@ const portField = ({
   changeMethods,
   getValidation,
   validationErrors,
+  db,
 }: FieldPropTypes) => (
-  <ValidatedInput
-    id="port"
-    name="port"
-    required={required}
-    validationMethods={{ onBlur: getValidation }}
-    errorMessage={validationErrors?.port}
-    placeholder="e.g. 5432"
-    className="form-group-w-50"
-    label="Port"
-    onChange={changeMethods.onParametersChange}
-  />
+  <>
+    <ValidatedInput
+      id="port"
+      name="port"
+      type="number"
+      required={required}
+      value={db?.parameters?.port as number}
+      validationMethods={{ onBlur: getValidation }}
+      errorMessage={validationErrors?.port}
+      placeholder="e.g. 5432"
+      className="form-group-w-50"
+      label="Port"
+      onChange={changeMethods.onParametersChange}
+    />
+  </>
 );
 const databaseField = ({
   required,
   changeMethods,
   getValidation,
   validationErrors,
+  db,
 }: FieldPropTypes) => (
   <ValidatedInput
     id="database"
     name="database"
     required={required}
+    value={db?.parameters?.database}
     validationMethods={{ onBlur: getValidation }}
     errorMessage={validationErrors?.database}
     placeholder="e.g. world_population"
     label="Database name"
     onChange={changeMethods.onParametersChange}
-    helpText="Copy the name of the PostgreSQL database you are trying to connect to."
+    helpText="Copy the name of the  database you are trying to connect to."
```

Comment:
Same here, `t()`.

Reply:
I _think_ I got them all in [`this commit`](https://github.com/apache/superset/pull/15463/commits/b8be120b60f52982f7135c222828b0426c322faf).